pkg/client/util/util_test.go

```diff
@@ -45,3 +49,56 @@ func TestJoinRepoImage(t *testing.T) {
 		})
 	}
 }
+
+func TestOSArchFromTag(t *testing.T) {
+	tests := map[string]struct {
+		tag     string
+		expOS   api.OS
+		expArch api.Architecture
+	}{
+		"empty tag should return empty OS and Arch": {
+			tag:     "",
+			expOS:   "",
+			expArch: "",
+		},
+		"tag with only OS should return correct OS and empty Arch": {
+			tag:     "v1.0.0-linux",
+			expOS:   "linux",
+			expArch: "",
+		},
+		"tag with only Arch should return linux OS and correct Arch": {
+			tag:     "v1.0.0-amd64",
+			expOS:   "linux",
+			expArch: "amd64",
+		},
+		"tag with OS and Arch should return both correctly": {
+			tag:     "v1.0.0-linux-amd64",
+			expOS:   "linux",
+			expArch: "amd64",
+		},
+		"tag with multiple segments should return correct OS and Arch": {
+			tag:     "v1.0.0-linux-arm64",
+			expOS:   "linux",
+			expArch: "arm64",
+		},
```

Comment:
these are the same test?

Reply:
Similar, for sure! but not identical, one was arm other amd..

```suggestion
		"tag with OS and Arch should return both correctly": {
			tag:     "v1.0.0-linux-amd64",
			expOS:   "linux",
			expArch: "amd64",
		},
```